src/main/scala/chisel3/util/circt/PlusArgsValue.scala

```diff
@@ -0,0 +1,46 @@
+// SPDX-License-Identifier: Apache-2.0
+
+package chisel3.util.circt
+
+import chisel3._
+import chisel3.experimental.{annotate, ChiselAnnotation, ExtModule, FlatIO}
+import chisel3.internal.Builder
+
+import circt.Intrinsic
+
+/** Create a module which generates a verilog $plusargs$value.  This returns a
+  * value as indicated by the format string and a flag for whether the value
+  * was found.
+  */
+private class PlusArgsValueIntrinsic[T <: Data](gen: T, str: String) extends ExtModule(Map("FORMAT" -> str)) {
+  val io = FlatIO(new Bundle {
+    val found = Output(UInt(1.W))
+    val result = Output(gen)
+  })
+  annotate(new ChiselAnnotation {
+    override def toFirrtl =
+      Intrinsic(toTarget, "circt.plusargs.value")
+  })
+  override val desiredName = "PlusArgsValue_" + Builder.idGen.next
+}
+
+object PlusArgsValue {
+
+  /** Creates an intrinsic which calls $test$plusargs.
+    *
+    * @example {{{
+    * b := PlusArgsValue(UInt<32.W>, "FOO=%d")
```

Comment:
```suggestion
    * b := PlusArgsValue(UInt(32.W), "FOO=%d")
```

Reply:
what is the point of exposing to the user the `Foo=%d` vs just the string foo? is it to allow different string formats?

Reply:
yes, you can have fancy string formats in verilog.